### PR TITLE
Preserve functions when updating process

### PR DESCRIPTION
**Pull request by dimitry-**
Bug: http://b/152854934

### CODE REVIEW

**Comment by dimitry-** [comment]
modules are refreshed but yes we would want to preserve loaded flag..

File: OrbitGl/App.cpp [hunk 1]
```diff
@@ -1049,6 +1049,29 @@ std::shared_ptr<Process> OrbitApp::FindProcessByPid(uint32_t pid) {
...
+    std::shared_ptr<Process> old_process = it->second;
+    std::map<uint64_t, std::shared_ptr<Module>>& old_modules =
+		old_process->GetModules();
+    std::map<uint64_t, std::shared_ptr<Module>>& new_modules =
+        process->GetModules();
+
+	// Move 'loaded' flag to new modules if possible.
+	for (auto module_entry : old_modules) {
+      uint64_t key = module_entry.first;
+      auto it = new_modules.find(key);
```
Comment (dpallotti) [inline OrbitGl/App.cpp:1067]: This `it` shadows the `it` in the outer scope, maybe rename both to more descriptive names.

File: OrbitGl/App.cpp [hunk 2]
```diff
@@ -1049,6 +1049,29 @@ std::shared_ptr<Process> OrbitApp::FindProcessByPid(uint32_t pid) {
...
+    // separate structures.
+    std::shared_ptr<Process> old_process = it->second;
+    std::map<uint64_t, std::shared_ptr<Module>>& old_modules =
+		old_process->GetModules();
+    std::map<uint64_t, std::shared_ptr<Module>>& new_modules =
+        process->GetModules();
+
+	// Move 'loaded' flag to new modules if possible.
+	for (auto module_entry : old_modules) {
+      uint64_t key = module_entry.first;
```
Comment (dpallotti) [inline OrbitGl/App.cpp:1066]: Indentation is wrong.

File: OrbitGl/App.cpp (hunk 1 above)
Comment (dimitry-) [reply OrbitGl/App.cpp]: done

File: OrbitGl/App.cpp (hunk 2 above)
Comment (dimitry-) [reply OrbitGl/App.cpp:1066]: done.

File: OrbitGl/App.cpp [hunk 3]
```diff
@@ -1049,6 +1049,29 @@ std::shared_ptr<Process> OrbitApp::FindProcessByPid(uint32_t pid) {
 
 void OrbitApp::UpdateProcess(const std::shared_ptr<Process>& process) {
   absl::MutexLock lock(&process_map_mutex_);
+  // We want to move function map to new process object
+  auto it = process_map_.find(process->GetID());
```
Comment (dpallotti) [inline OrbitGl/App.cpp:1053]: Maybe rename this as well, maybe to `old_process_it`?

Comment (dimitry-) [reply OrbitGl/App.cpp]: done

**Pull request merged**